lib/redshift/client/connection.rb

```diff
@@ -11,16 +11,17 @@ def initialize(configuration)
         @original = PG.connect(configuration.params)
       end
 
-      def_delegators \
-        :@original,
-        :exec,
-        :exec_params,
-        :escape,
-        :escape_string,
-        :escape_literal,
-        :close,
-        :transaction,
-        :quote_ident
+      def method_missing(method_name, *args, &block)
+        if @original.respond_to?(method_name)
+          @original.send(method_name, *args, &block)
+        else
+          super
+        end
+      end
+
+      def respond_to_missing?(method_name, include_private = false)
```

Comment:
Where will we use this method? Can you give some super simplified context for these methods?

Reply:
This is something similar as the method missing just this is responsible for a class `respond_to?` method. Now for every missing method runby will also asks the connection whether it is able to answer for that or not. These 2 methods usually works together or the respond_to will lie!